pkg/protoassert/assert.go

```diff
@@ -0,0 +1,82 @@
+package protoassert
+
+import (
+	"testing"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+)
+
+func Equal(t *testing.T, expected, actual proto.Message, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	e, err := toJson(expected)
+	require.NoError(t, err)
+	a, err := toJson(actual)
+	require.NoError(t, err)
+	return assert.JSONEq(t, e, a, msgAndArgs)
+}
+
+func SlicesEqual[T proto.Message](t *testing.T, expected, actual []T, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	areEqual := assert.Len(t, actual, len(expected), msgAndArgs)
```

Comment:
nitpick: Since we will output `msgAndArgs` with, there is no need to pass it to `assert.Len` or `Equal`
```
	if !areEqual {
		t.Log(msgAndArgs...)
	}
```

**EDIT:** I see that for Map functions message is not passed to other "assert" calls. We should do the same here.